sdk/src/client/api/high_level.rs

```diff
@@ -28,27 +28,32 @@ use crate::{
 impl Client {
     /// Get the inputs of a transaction for the given transaction id.
     pub async fn inputs_from_transaction_id(&self, transaction_id: &TransactionId) -> Result<Vec<OutputWithMetadata>> {
-        let block = self.get_included_block(transaction_id).await?;
-
-        let inputs = match block.payload() {
-            Some(Payload::Transaction(t)) => t.essence().inputs(),
-            _ => {
-                unreachable!()
-            }
-        };
-
-        let input_ids = inputs
-            .iter()
-            .map(|i| match i {
-                Input::Utxo(input) => *input.output_id(),
+        let wrapper = self.get_included_block(transaction_id).await?;
+
+        if let Block::Basic(block) = wrapper.block() {
+            let inputs = match block.payload() {
+                Some(Payload::Transaction(t)) => t.essence().inputs(),
+                _ => return Err(Error::MissingTransactionPayload),
+            };
```

Comment:
consider replacing that match with `if let ... else`

Reply:
if let is more verbose but if you want, sure

Reply:
Done